docroot/modules/custom/hs_field_helpers/src/EventSubscriber/EventSubscriber.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Drupal\hs_field_helpers\EventSubscriber;
+
+use Drupal\Core\Link;
+use Drupal\Core\Menu\MenuActiveTrailInterface;
+use Drupal\Core\Menu\MenuLinkManagerInterface;
+use Drupal\layout_builder\Event\SectionComponentBuildRenderArrayEvent;
+use Drupal\layout_builder\LayoutBuilderEvents;
+use Drupal\menu_block\Plugin\Block\MenuBlock;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class EventSubscriber
+ *
+ * @package Drupal\hs_field_helpers\EventSubscriber
+ */
+class EventSubscriber implements EventSubscriberInterface {
+
+  /**
+   * Active trail service.
+   *
+   * @var \Drupal\Core\Menu\MenuActiveTrailInterface
+   */
+  protected $activeTrail;
+
+  /**
+   * Menu link manager service.
+   *
+   * @var \Drupal\Core\Menu\MenuLinkManagerInterface
+   */
+  protected $linkManager;
+
+  /**
+   * EventSubscriber constructor.
+   *
+   * @param \Drupal\Core\Menu\MenuActiveTrailInterface $active_trail
+   *   Active trail service.
+   * @param \Drupal\Core\Menu\MenuLinkManagerInterface $link_manager
+   *   Menu link manager service.
+   */
+  public function __construct(MenuActiveTrailInterface $active_trail, MenuLinkManagerInterface $link_manager) {
+    $this->activeTrail = $active_trail;
+    $this->linkManager = $link_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events[LayoutBuilderEvents::SECTION_COMPONENT_BUILD_RENDER_ARRAY][] = [
+      'onLayoutBuilderRender',
+      1,
+    ];
+    return $events;
+  }
+
+  /**
+   * Set the label of the block for layout builder blocks.
+   *
+   * @param \Drupal\layout_builder\Event\SectionComponentBuildRenderArrayEvent $event
+   *   Layout builder render array event.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function onLayoutBuilderRender(SectionComponentBuildRenderArrayEvent $event) {
+    if ($event->getPlugin() instanceof MenuBlock) {
```

Comment:
That's a bummer you had to account for empty tags.

Reply:
I actually might not need to with the use of this patch https://www.drupal.org/project/menu_block/issues/2809699#comment-12604349

Reply:
I removed that part and change the patch for menu_block module to one that checks for the empty block.

Reply:
Nice.